kiwi/shareddata.h

```diff
@@ -16,22 +16,25 @@ class SharedData
 public:
     SharedData() : m_refcount(0) {}
 
-    SharedData(const SharedData &other) : m_refcount(0) {}
+    SharedData(const SharedData &other) = delete;
+
+    SharedData(SharedData&& other) = delete;
 
     int m_refcount;
 
-private:
-    SharedData &operator=(const SharedData &other);
+    SharedData &operator=(const SharedData &other) = delete;
+    
+    SharedData &operator=(SharedData&& other) = delete;
 };
 
 template <typename T>
 class SharedDataPtr
 {
 
 public:
-    typedef T Type;
+    using Type = T;
 
-    SharedDataPtr() : m_data(0) {}
+    SharedDataPtr() : m_data(nullptr) {}
```

Comment:
Again, are we sure these two are equivalent?

Reply:
Yes it is equivalent, since `m_data` is a raw pointer.